### PR TITLE
It's the Cloud SDK that is not found, not the Cloud SDK path.

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis also uses a more specific exception

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java [hunk 1]
```diff
@@ -641,8 +641,8 @@ private Path discoverSdkPath() {
               + ": exception thrown when searching for Google Cloud SDK", ex);
         }
       }
-      throw new AppEngineException("Google Cloud SDK path was not provided and could not be"
-          + " found in any known install locations.");
+      throw new CloudSdkNotFoundException("The Google Cloud SDK could not be found in the default"
+          + " locations and no path was provided.");
```
Comment (briandealwis) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: Not sure if _default_ is correct here as the different resolvers  may have searched in more than the default locations.

_The Google Cloud SDK could not be located_?

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: how about default --> customary?

Comment (briandealwis) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: "customary" works for me.  I was unsure about _no path was provided_ but that seems covered with `GOOGLE_CLOUD_SDK_HOME` and the preferences.

**Comment by elharo** [comment]
customary it is

**Pull request merged**